kimchi/src/circuits/gate.rs

```diff
@@ -422,13 +423,36 @@ impl GateType {
             chacha_final_where.insert((ChaChaFinal, *r));
         }
 
+        let lookup_gate_pattern = (0..3)
+            .map(|i| {
+                // each row represents an XOR operation
+                // where l XOR r = o
+                //
+                // 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14
+                // - i v - - - - - - - -  -  -  -  -
+                // - - - i v - - - - - -  -  -  -  -
+                // - - - - - i v - - - -  -  -  -  -
+                let index = curr_row(2 * i + 1);
+                let value = curr_row(2 * i + 2);
+                let l = |loc: LocalPosition| SingleLookup {
+                    value: vec![(F::one(), loc)],
+                };
+                JointLookup {
+                    table_id: 1,
```

Comment:
a few things:

* use a hardcoded const like `XOR_TABLE_ID`
* why is table_id set to `1`? If it is supposed to represent the runtime table?

Reply:
The next PR in the chain (#400) does it properly, using the first position in the witness. This is really a temporary placeholder until then, since those changes are orthogonal. I chose 1 because it makes it easier to write a reasonable test without requiring the changes in #400.